src/provider.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import * as React from 'react';
+import { checkSessionAction } from './actions.js';
+
+export const Provider = ({ children }: React.PropsWithChildren) => {
+  React.useEffect(() => {
+    const handleVisibilityChange = async () => {
+      if (document.visibilityState === 'visible') {
+        try {
+          const hasSession = await checkSessionAction();
+          if (!hasSession) {
+            throw new Error('Session expired');
+          }
+        } catch (error) {
+          window.location.reload();
```

Comment:
Should this be configurable? What if a developer wants to take an action other than reloading the page?